File: Backend/renderer/src/main/java/net/chrisparton/sparkled/renderer/fill/FillFunctions.java
package net.chrisparton.sparkled.renderer.fill;

import net.chrisparton.sparkled.model.animation.fill.FillTypeCode;
import net.chrisparton.sparkled.renderer.fill.function.RainbowFill;
import net.chrisparton.sparkled.renderer.fill.function.SolidFill;

import java.util.HashMap;
import java.util.Map;

public class FillFunctions {

    private static Map<FillTypeCode, FillFunction> functions = new HashMap<>();

    static {
        functions.put(FillTypeCode.RAINBOW, new RainbowFill());
        functions.put(FillTypeCode.SOLID, new SolidFill());
    }

    public static FillFunction get(FillTypeCode fillType) {
        return functions.get(fillType);
    }
}

File: renderer/src/main/kotlin/io/sparkled/renderer/fill/FillFunctions.kt
package io.sparkled.renderer.fill

import io.sparkled.model.animation.fill.FillTypeCode
import io.sparkled.renderer.fill.function.RainbowFill
import io.sparkled.renderer.fill.function.SolidFill
import java.util.HashMap

object FillFunctions {

    private val functions = HashMap<FillTypeCode, FillFunction>()

    init {
        functions[FillTypeCode.RAINBOW] = RainbowFill()
        functions[FillTypeCode.SOLID] = SolidFill()
    }

    operator fun get(fillType: FillTypeCode): FillFunction {
        return functions[fillType]!!
    }
}